### PR TITLE
UI Wrangler Fixes

### PR DESCRIPTION
**Pull request by elfenheart**
JIRAs: 
- https://issues.cask.co/browse/WRANGLER-31
- https://issues.cask.co/browse/WRANGLER-38
- https://issues.cask.co/browse/WRANGLER-27
- https://issues.cask.co/browse/WRANGLER-3

Issues Fixed:
- Visualization store properly reset when component unmount
- Clicking anywhere on the delimiter line will activate the focus
- Explore Tab Item initialization of selected field
- Add Warning modal when leaving Wrangler modal in Pipeline

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/Wrangler/wrangler-modal-ctrl.js [hunk 1]
```diff
@@ -15,12 +15,19 @@
...
     this.node = rPlugin;
     this.$uibModalInstance = $uibModalInstance;
     this.EventPipe = EventPipe;
 
     this.applyToHydrator = this.applyToHydrator.bind(this);
+
+    $scope.$on('modal.closing', (e, reason) => {
+      if (reason === 'ADD_TO_PIPELINE') { return; }
+
+      let shouldClose = confirm('Are you sure you want to exit Wrangler?');
```
Comment (bdmogal) [inline cdap-ui/app/hydrator/controllers/create/Wrangler/wrangler-modal-ctrl.js:28]: i18n?

Comment (ajainarayanan) [reply cdap-ui/app/hydrator/controllers/create/Wrangler/wrangler-modal-ctrl.js:28]: Unfortunately we don't have i18n in angular part of the code

Comment (bdmogal) [reply cdap-ui/app/hydrator/controllers/create/Wrangler/wrangler-modal-ctrl.js:28]: oh this is hydrator. nvm.

**Comment by bdmogal** [comment]
Verified these fixes. LGTM. May want to wait for @ajainarayanan or @tonybach to review the code, but the fixes work fine.

**Comment by ajainarayanan** [comment]
👍 LGTM

**Comment by elfenheart** [comment]
Build: http://builds.cask.co/browse/CDAP-DRC5811-2

**Pull request merged**